### PR TITLE
DOC: BIDS filter input example

### PR DESCRIPTION
**Pull request by utooley**
#2027, one stab at it.

Added a few lines to the Usage Notes as an example of input to `--bids-filter-file`. Took me a while of digging through pybids to figure out how to set the `config.json` up properly; I think that it would be useful to link to an example like that in the discussion of #1770, but couldn't figure out where to put it.

## Documentation that should be reviewed

Usage Notes - Positional arguments - bids-filter-file

### CODE REVIEW

**Comment by oesteban** [comment]
Could you rebase with upstream/master? The latest PR merge should fix the issues you are encountering (namely, dropping one of the flake8 extensions).

**Comment by utooley** [comment]
Sorry, I don't think I did this properly...I was having trouble with `rebase`, possibly because of a bug in OSX that kept me in an infinite loop, possibly because I haven't ever had to use it before and was trying to learn by doing. Please tell me if I should fix something!

**Comment by effigies** [comment]
Can you show `git remote -v`? It will make it easier to give rebase instructions.

**Comment by utooley** [comment]
```git remote -v
origin	git@github.com:utooley/fmriprep.git (fetch)
origin	git@github.com:utooley/fmriprep.git (push)
upstream	https://github.com/poldracklab/fmriprep.git (fetch)
upstream	https://github.com/poldracklab/fmriprep.git (push)```

I did :
```
git fetch upstream
git rebase upstream/master
```
Got caught in a merge conflict, resolved it, but then an infinite loop of "no changes" when trying to `git rebase --continue`. Apparently there's [this](https://stackoverflow.com/questions/21889741/git-rebase-continue-wont-work) which I've now fixed, but not sure if that was really the problem.

**Comment by effigies** [comment]
Looks like there's a merge commit. (See [network](https://github.com/poldracklab/fmriprep/network).)

Just `git rebase -i upstream/master` and set the plan to:

```
pick ba0af72
```

Save and quit, and it should go smoothly.

**Comment by utooley** [comment]
Alright, I think I fixed it--let me know if this is wrong! Took me a bit to figure out that I needed to force-push.

**Comment by effigies** [comment]
> Took me a bit to figure out that I needed to force-push.

Ah, sorry. Yes. I forgot about that. This looks like a good rebase.

**Comment by utooley** [comment]
Alright--I added a section to the FAQ, and a pointer in `parser.py`, though PEP8 doesn't seem to link how long the link is (I tried to break it into a new line in b4eb01b but it didn't seem to work properly?). Not sure what conventions for that are.

File: fmriprep/cli/parser.py [hunk 1]
```diff
@@ -94,10 +94,11 @@ def _bids_filter(value):
...
         type=_bids_filter, metavar='PATH',
-        help='a JSON file describing custom BIDS input filter using pybids '
-             '{<suffix>:{<entity>:<filter>,...},...} '
-             '(https://github.com/bids-standard/pybids/blob/master/bids/layout/config/bids.json)')
-
+        help="""\
+    a JSON file describing custom BIDS input filter using pybids.
+    For further details, please check out
+    https://fmriprep.readthedocs.io/en/%s/faq.html#how-do-I-select-only-certain-files-to-be-input-to-fMRIPrep""" % (currentv.base_version
+                                                 if is_release else 'latest'))
```
Comment (mgxd) [inline fmriprep/cli/parser.py]:
```suggestion
        help="a JSON file describing custom BIDS input filter using pybids. "
             "For further details, please check out "
             "https://fmriprep.readthedocs.io/en/%s/faq.html#"
             "how-do-I-select-only-certain-files-to-be-input-to-fMRIPrep" % (
                 currentv.base_version if is_release else 'latest'))
```

**Comment by utooley** [comment]
Thanks for the tip! I didn't think of formatting that way.

File: docs/faq.rst [hunk 2]
```diff
@@ -220,3 +220,27 @@ After getting the resources you'll need, you will just need to make sure your
 runtime environment is able to access the filesystem, at the location of your
 *TemplateFlow home* directory.
 If you are a Singularity user, please check out :ref:`singularity_tf`.
+
+How do I select only certain files as input to fMRIPrep?
+---------------------------------------------------------------------
+
+Using the `--bids-filter-file` flag, you can pass fMRIPrep a JSON file that
```
Comment (mgxd) [inline docs/faq.rst]: The two hyphens are currently being merged into a dash
https://12540-53175327-gh.circle-artifacts.com/0/tmp/src/fmriprep/docs/_build/html/faq.html#how-do-i-select-only-certain-files-as-input-to-fmriprep

this should fix it
```suggestion
Using the ``--bids-filter-file`` flag, you can pass fMRIPrep a JSON file that
```

Comment (utooley) [reply docs/faq.rst]: Wow, I thought I was detailed-oriented, but did _not_ catch that--thank you!

File: docs/faq.rst [hunk 3]
```diff
@@ -220,3 +220,27 @@ After getting the resources you'll need, you will just need to make sure your
 runtime environment is able to access the filesystem, at the location of your
 *TemplateFlow home* directory.
 If you are a Singularity user, please check out :ref:`singularity_tf`.
+
+How do I select only certain files to be input to fMRIPrep?
+---------------------------------------------------------------------
```
Comment (effigies) [inline docs/faq.rst]:
```suggestion
How do I select only certain files to be input to fMRIPrep?
-----------------------------------------------------------
```

File: docs/faq.rst [hunk 4]
```diff
@@ -220,3 +220,27 @@ After getting the resources you'll need, you will just need to make sure your
...
+
+Using the ``--bids-filter-file`` flag, you can pass fMRIPrep a JSON file that
+describes a custom BIDS filter using pyBIDS, with the syntax
+{<suffix>:{<entity>:<filter>,...},...}. For example: ::
```
Comment (effigies) [inline docs/faq.rst]:
```suggestion
Using the ``--bids-filter-file`` flag, you can pass fMRIPrep a JSON file that
describes a custom BIDS filter for selecting files with PyBIDS, with the syntax
``{<query>: {<entity>: <filter>, ...},...}``. For example::
```

File: docs/faq.rst [hunk 5]
```diff
@@ -220,3 +220,27 @@ After getting the resources you'll need, you will just need to make sure your
...
+      },
+      "bold": {
+          "datatype": "func",
+          "session": "02",
+          "suffix": "bold"
+      }
+  }
+
+For all possible filters, see the pyBIDS
+`config file <https://github.com/bids-standard/pybids/blob/master/bids/layout/config/bids.json>`__.
```
Comment (effigies) [inline docs/faq.rst]:
```suggestion
fMRIPrep uses the following queries, by default::

  {
    'fmap': {'datatype': 'fmap'},
    'bold': {'datatype': 'func', 'suffix': 'bold'},
    'sbref': {'datatype': 'func', 'suffix': 'sbref'},
    'flair': {'datatype': 'anat', 'suffix': 'FLAIR'},
    't2w': {'datatype': 'anat', 'suffix': 'T2w'},
    't1w': {'datatype': 'anat', 'suffix': 'T1w'},
    'roi': {'datatype': 'anat', 'suffix': 'roi'},
  }

Only modifications of these queries will have any effect. You may filter on any entity defined
in the the PyBIDS
`config file <https://github.com/bids-standard/pybids/blob/master/bids/layout/config/bids.json>`__.
```

File: fmriprep/cli/run.py [hunk 6]
```diff
@@ -79,9 +79,11 @@ def get_parser():
     #                     help='select a specific run to be processed')
     g_bids.add_argument(
         '--bids-filter-file', action='store', type=Path, metavar='PATH',
-        help='a JSON file describing custom BIDS input filter using pybids '
-             '{<suffix>:{<entity>:<filter>,...},...} '
-             '(https://github.com/bids-standard/pybids/blob/master/bids/layout/config/bids.json)')
+        help="a JSON file describing custom BIDS input filter using pybids. "
```
Comment (effigies) [inline fmriprep/cli/run.py]:
```suggestion
        help="a JSON file describing custom BIDS input filters using PyBIDS. "
```

**Pull request merged**